lib/webpacker/manifest.rb

```diff
@@ -13,32 +13,95 @@ def file_path
       Webpacker::Configuration.manifest_path
     end
 
+    # Throws an error if the file is not found. If Configuration.compile? then compilation is invoked
+    # the file is missing.
+    # React on Rails users will need to set Configuration.compile? to false as compilation is configured
+    # in the package.json for React on Rails.
     def lookup(name)
       if Webpacker::Configuration.compile?
         compile_and_find!(name)
       else
+        # Since load_instance is a no-op for production, this should be fine.
+        # If we're using `webpack -w` to recompile when files changes, we need
+        # to call this before find!
+        load_instance
```

Comment:
Not exactly confidence inspiring that something "should be fine" 😄. We should explain clearly here why/when the #load needs to be called prior to find!